src/NUnitFramework/framework/Internal/Results/TestResult.cs

```diff
@@ -498,24 +498,15 @@ public void RecordException(Exception ex)
             if (ex is NUnitException || ex is TargetInvocationException)
                 ex = ex.InnerException;
 
-            if (ex is AssertionException)
-            {
-                AssertionResults.Add(new AssertionResult(AssertionStatus.Failed, ex.Message, StackFilter.DefaultFilter.Filter(ex.StackTrace)));
-            }
-
             if (ex is ResultStateException)
-            {
-                string message = ex is MultipleAssertException
-                    ? CreateLegacyFailureMessage()
-                    : ex.Message;
-
-                string stackTrace = StackFilter.DefaultFilter.Filter(ex.StackTrace);
-
-                SetResult(((ResultStateException)ex).ResultState, message, stackTrace);
-            }
+                SetResult(
+                    ((ResultStateException)ex).ResultState,
+                    ex.Message, 
+                    StackFilter.DefaultFilter.Filter(ex.StackTrace));
 #if !NETSTANDARD1_6
             else if (ex is System.Threading.ThreadAbortException)
-                SetResult(ResultState.Cancelled,
+                SetResult(
+                    ResultState.Cancelled,
```

Comment:
Unrelated formatting change, I'd suggest not performing it as part of this PR to avoid behavior+refactor changes in the same boat.